xformers/components/nvfuser/bias_act_dropout.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import logging
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import torch
+import torch.nn as nn
+from functorch.compile import memory_efficient_fusion
+
+from xformers.components import Activation, build_activation
+from xformers.components.nvfuser import Fused, FusedConfig, register_fused
+
+
+@dataclass
+class FusedBiasActivationDropoutConfig(FusedConfig):
+    p: float
+    activation: Activation
+    bias_shape: Optional[int]
+
+
+def _fn(
+    activation: Activation, prob: float, x: torch.Tensor, bias: torch.Tensor
+) -> torch.Tensor:
+    if bias is not None:
+        x = torch.add(x, bias)
+    y = activation(x)
+    return torch.nn.functional.dropout(y, prob) if prob > 0.0 else y
+
+
+@register_fused("fused_bias_activation_dropout", FusedBiasActivationDropoutConfig)
+class FusedBiasActivationDropout(Fused):
```

Comment:
You can look at triton fused layer norm as an example [here](https://github.com/facebookresearch/xformers/blob/main/xformers/triton/layer_norm.py#L166)